### PR TITLE
Sign in with Metamask feature

### PR DESCRIPTION
**Pull request by zediogoviana**
Why:
 - In order to make this project more interactive and accessible, we decided to add a sign-in with Metamask for a pre-configured wallet User.

How:
 - This pre-configured User Address should be inserted when running the seeds, or directly created in the console if the seeds are not run.
 - Only users with access to addresses in the DB can log in. This means that there is no "sign up with metamask" flow.
 - This PR introduces a live view component that represents the Metamask button and all its logic, which works together with a JS file called `metamask.js`. Both files pass events around to the full thing reactive and update live.
 - Each time a user tries to log in, a new `Nonce` is generated and the signature check is against that nonce. `:ex_web3_ec_recover` takes care of the signature verification. If the addresses match, then the user is logged in.

Screenshot 2023-03-01 at 12 34 32

### CODE REVIEW

File: assets/js/hooks/metamask.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    }
+}
+
+export const Metamask = {
+    mounted() {
+        let signer = web3Provider.getSigner()
+
+        window.addEventListener('load', async () => {
+            init();
+            let address
```
Comment (DavideSilva) [inline assets/js/hooks/metamask.js:19]: I'm struggling to understand why this is needed. It seems it's nothing is ever assigned to this variable

File: assets/js/hooks/metamask.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+const web3Provider = new ethers.providers.Web3Provider(window.ethereum)
+
+function init() {
+    // Check that the web page is run in a secure context,
+    // as otherwise MetaMask won't be available
+    if (location.hostname !== "localhost" && location.protocol !== 'https:') {
+        console.log("FAILING AT HTTPS")
+    }
+}
```
Comment (DavideSilva) [inline assets/js/hooks/metamask.js:11]: This is not really checking that the page is in a secure context, it's just printing out stuff to the console. Are you planning to refactor this later on?

**Review by DavideSilva — CHANGES_REQUESTED** [review]

File: assets/js/hooks/metamask.js (hunk 1 above)
Comment (zediogoviana) [reply assets/js/hooks/metamask.js:19]: I forgot this here when cleaning up the code. You're right, it's doing nothing...

File: assets/js/hooks/metamask.js (hunk 2 above)
Comment (zediogoviana) [reply assets/js/hooks/metamask.js:11]: No, this was actually a piece of code from the original repo where I got the base. I can remove it, though...

**Review by DavideSilva — APPROVED** [review]
:shipit:

**Pull request merged**